apps/android_rpc/app/src/main/jni/tvm_runtime.h

```diff
@@ -6,6 +6,9 @@
 #include <sys/stat.h>
 #include <fstream>
 
+#define DMLC_LOG_CUSTOMIZE 1
+#define DMLC_LOG_BEFORE_THROW 1
```

Comment:
purpose of these macros ?

Reply:
`DMLC_LOG_CUSTOMIZE` is what tells TVM to use the `CustomMessage` class for logging instead of `LogMessage`. This requires us to define an implementation for the static function `dmlc::CustomLogMessage::Log`, which will be invoked for every log that TVM records. This is what allows us to pass the log messages to one of the android logging functions so that they will show up in logcat.

`DMLC_LOG_BEFORE_THROW` ensures that the log function is invoked before throwing in `LogMessageFatal`, so that again we have an opportunity to pass the message on to logcat.

see `dmlc-core/include/dmlc/logging.h` for the implementation of both of these